src/Swarm/TUI/Controller.hs

```diff
@@ -390,7 +390,7 @@ handleREPLEvent s (VtyEvent (V.EvKey (V.KChar 'c') [V.MCtrl])) =
       & gameState . robotMap . ix "base" . machine .~ idleMachine
 handleREPLEvent s (VtyEvent (V.EvKey V.KEnter [])) =
   if not $ s ^. gameState . replWorking
-    then case processTerm' topCtx topCapCtx entry of
+    then case processTerm' topTypeCtx topCapCtx entry of
```

Comment:
What do you think about changing `processTerm'` to just take a `RobotCtx` instead of separate type and cap contexts?  It's true that it doesn't really need the `valCtx` but I don't think it would hurt anything.  It means this code would be simpler because we wouldn't have to project out all the individual contexts in order to pass them to `processTerm'`.

Reply:
I think that we should keep `processTerm'` the same., firstly because projecting the two individual contexts is not really that bad.

But secondly and more importantly is that while `RobotContext` fields' types are the same as the arguments to `processTerm'`, I think they mean different things. `RobotContext` is stores information of all the definitions for the duration of the lifetime of a robot but `processTerm'` is more general and can start with any context. I feel that the current types make this more evident.